### PR TITLE
fixed reported issues with some untested code paths throwing errors. …

### PR DESCRIPTION
**Pull request by melifluos**
…All tests passing

### CODE REVIEW

File: src/best_params.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
                     }
```
Comment (JRowbottomGit) [inline src/best_params.py:8]: no heterophilic dataset params

File: src/function_transformer_attention.py [hunk 2]
```diff
@@ -186,13 +186,13 @@ def forward(self, x, edge):
       src = q[edge[0, :], :, :]
       dst_k = k[edge[1, :], :, :]
 
-    if self.opt['attention_type'] == "scaled_dot":
+    if not self.opt['beltrami'] and self.opt['attention_type'] == "exp_kernel":
+      prods = (torch.sum((src - dst_k) ** 2, dim=1) / (2 * self.lengthscale ** 2))
+    elif self.opt['attention_type'] == "scaled_dot":
       prods = torch.sum(src * dst_k, dim=1) / np.sqrt(self.d_k)
```
Comment (JRowbottomGit) [inline src/function_transformer_attention.py:192]: this should be:
prods = self.output_var ** 2 * torch.exp(-(torch.sum((src - dst_k) ** 2, dim=1) / (2 * self.lengthscale ** 2)))

will also need to make changes to init, I'll fix after this PR

**Review by JRowbottomGit — APPROVED** [review]
One question about whether we include the heterophilic script as no heterophilic best params included.

One fix I will do now on exponential kernal type attention for non-beltrami path
